src/nvme/types.h

```diff
@@ -9007,4 +9025,350 @@ struct nvme_ns_mgmt_host_sw_specified {
+	NVME_LM_TRACK_SEND_SELECT_SHIFT			= 0,
+	NVME_LM_TRACK_SEND_SELECT_MASK			= 16,
+	NVME_LM_LACT_STOP_LOGGING			= 0,
+	NVME_LM_LACT_START_LOGGING			= 1,
+	NVME_LM_LACT_MASK				= 0xf,
+	NVME_LM_LACT_SHIFT				= 16,
+};
+
+/**
+ * enum nvme_lm_migration_send_fields - Migration Send command fields
+ *
+ * @NVME_LM_SUSPEND:			Migration Send - Suspend
+ * @NVME_LM_RESUME:			Migration Send - Resume
+ * @NVME_LM_SET_CONTROLLER_STATE:	Migration Send - Set Controller State
+ * @NVME_LM_MIGRATION_SEND_SEL_SHIFT:	Shift amount to set Select (SEL) field
+ * @NVME_LM_MIGRATION_SEND_SEL_MASK:	Mask to set SEL field
+ * @NVME_LM_MIGRATION_SEND_CNTLID_SHIFT: Shift amount to set Controller ID (CNTLID)
+ * @NVME_LM_MIGRATION_SEND_CNTLID_MASK:	Mask to set CNTLID
+ * @NVME_LM_MIGRATION_SEND_UIDX_SHIFT:	Shift to set UUID Index (UIDX)
+ * @NVME_LM_MIGRATION_SEND_UIDX_MASK:	Mask to set UIDX
+ * @NVME_LM_DUDMQ:			Delete User Data Migration Queue (DUDMQ)
+ * @NVME_LM_DUDMQ_SHIFT:		Shift amount to set DUDMQ
+ * @NVME_LM_DUDMQ_MASK:			Mask to set DUDMQ
+ * @NVME_LM_STYPE_NOTIFICATION:		Suspend Type (STYPE) - Suspend Notification
+ * @NVME_LM_STYPE_SUSPEND:		Suspend Type (STYPE) - Suspend
+ * @NVME_LM_STYPE_SHIFT:		Shift amount to set STYPE
+ * @NVME_LM_STYPE_MASK:			Mask to set STYPE
+ * @NVME_LM_SEQIND_NOT_FIRST_NOT_LAST:	This command is not the first or last of a sequence two or
+ *					more Migration Send commands with this management operation
+ *					used to transfer the controller state from host to
+ *					controller
+ * @NVME_LM_SEQIND_FIRST:		This command is the first of a sequence of two or more
+ *					Migration Send commands
+ * @NVME_LM_SEQIND_LAST:		This command is the last command of a sequence of two or
+ *					more Migration Send commands
+ * @NVME_LM_SEQIND_ENTIRE:		This Migration Send command is the only command and contains
+ *					the entire controller state for this management operation
+ * @NVME_LM_SEQIND_SHIFT:		Shift amount to set SEQIND
+ * @NVME_LM_SEQIND_MASK:		Mask to set SEQIND
+ * @NVME_LM_MIGRATION_SEND_CSUUIDI_SHIFT: Shift amount to set Controller State UUID Index (CSUUIDI)
+ * @NVME_LM_MIGRATION_SEND_CSUUIDI_MASK: Mask to set CSUUIDI
+ * @NVME_LM_MIGRATION_SEND_CSVI_SHIFT:	Shift amount to set Controller State Version Index (CSVI)
+ * @NVME_LM_MIGRATION_SEND_CSVI_MASK:	Mask to set CSVI
+ */
+enum nvme_lm_migration_send_fields {
+	NVME_LM_SUSPEND				= 0,
+	NVME_LM_RESUME				= 1,
+	NVME_LM_SET_CONTROLLER_STATE		= 2,
+	NVME_LM_MIGRATION_SEND_SEL_SHIFT	= 0,
+	NVME_LM_MIGRATION_SEND_SEL_MASK		= 0xff,
+	NVME_LM_MIGRATION_SEND_CNTLID_SHIFT	= 0,
+	NVME_LM_MIGRATION_SEND_CNTLID_MASK	= 0xffff,
+	NVME_LM_MIGRATION_SEND_UIDX_SHIFT	= 0,
+	NVME_LM_MIGRATION_SEND_UIDX_MASK	= 0x7f,
+
+	/* Migration Send - Suspend */
+	NVME_LM_DUDMQ				= 1,
+	NVME_LM_DUDMQ_SHIFT			= 31,
+	NVME_LM_DUDMQ_MASK			= 0x1,
+	NVME_LM_STYPE_NOTIFICATION		= 0,
+	NVME_LM_STYPE_SUSPEND			= 1,
+	NVME_LM_STYPE_SHIFT			= 16,
+	NVME_LM_STYPE_MASK			= 0xff,
+
```

Comment:
What about `CNTLID`?

Reply:
For Migration Send, CNTLID is common to all SEL types. See `NVME_LM_MIGRATION_SEND_CNTLID_SHIFT` above

Reply:
I think it would make sense to keep it KISS and follow the spec pretty close. The long time plan is to generate these headers directly from the spec and having a clever trick here will make any transition to generated header more painful than it needs to be.

Reply:
Got it. I broke CNTLID out into per SEL-type definitions.